include/range/v3/action/push_back.hpp

```diff
@@ -78,17 +78,33 @@ namespace ranges
             {
                 return bind_back(push_back, static_cast<T &&>(val));
             }
+#ifdef RANGES_WORKAROUND_MSVC_OLD_LAMBDA
             template<typename T, std::size_t N>
-            static auto bind(push_back_fn push_back, T (&val)[N])
+            struct lamduh
             {
-                return bind_back(
-                    [push_back](auto && rng, T(*val)[N])
-                        -> invoke_result_t<push_back_fn, decltype(rng), T(&)[N]> {
-                        return push_back(static_cast<decltype(rng)>(rng), *val);
-                    },
-                    &val);
-            }
+                T (&val_)[N];
 
+                template<typename Rng>
+                auto operator()(Rng && rng) const
+                    -> invoke_result_t<push_back_fn, Rng, T(&)[N]> {
+                    return push_back_fn{}(static_cast<Rng &&>(rng), val_);
+                }
+            };
+            template<typename T, std::size_t N>
+            static lamduh<T, N> bind(push_back_fn, T (&val)[N])
+            {
+                return {val};
+            }
+#else // ^^^ workaround / no workaround vvv
+            template<typename T, std::size_t N>
+            static auto bind(push_back_fn, T (&val)[N])
+            {
+                return [&val](auto && rng)
+                    -> invoke_result_t<push_back_fn, decltype(rng), T(&)[N]> {
+                    return push_back_fn{}(static_cast<decltype(rng)>(rng), val);
```

Comment:
@ericniebler You may want to look at this "no workaround" bit. I'm flattening "generic lambda that captures some stuff wrapped in `bind_back` that captures some other stuff" into "generic lambda that captures all the stuff" for throughput. I don't *think* I've lost any semantic value here - did you have an ulterior motive for the multiple layers of function wrappers that I'm missing?

Reply:
I think this is fine. `bind_back::operator()` has overloads for `&&`, `&`, and `const &`; none of which matters when the only captures are reference captures.